### PR TITLE
Adding CTR to SCDF

### PR DESCRIPTION
**Pull request by cppwfs**
Resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/3779
Resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/3780

### CODE REVIEW

File: src/kubernetes/server/server-deployment.yaml [hunk 1]
```diff
@@ -54,6 +54,8 @@ spec:
           value: 'true'
         - name: SPRING_CLOUD_DATAFLOW_FEATURES_SCHEDULES_ENABLED
           value: 'true'
+        - name: SPRING_CLOUD_DATAFLOW_TASK_COMPOSED_TASK_RUNNER_URI
+          value: 'docker://springcloud/spring-cloud-dataflow-composed-task-runner:2.5.0.BUILD-SNAPSHOT'
```
Comment (ilayaperumalg) [inline src/kubernetes/server/server-deployment.yaml:58]: We would need to use the template files to update the version. Similar to what is done for the data flow server version update as in [here](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/src/templates/kubernetes/server/server-deployment.yaml#L19)

**Comment by ilayaperumalg** [comment]
LGTM except for the K8s template file change.

**Comment by cppwfs** [comment]
Before merge a run needs to occur without tests so that the CTR bits can be built and pushed to repo and docker.
Also need to add the docker bits to bamboo.

File: spring-cloud-dataflow-composed-task-runner/pom.xml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>org.springframework.cloud</groupId>
+		<artifactId>spring-cloud-dataflow-parent</artifactId>
+		<version>2.5.0.BUILD-SNAPSHOT</version>
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-composed-task-runner/pom.xml:23]: Given that `master` is changed to use `2.6.0.BUILD-SNAPSHOT`, this needs to be changed. I can fix it when merging.

File: src/kubernetes/server/server-deployment.yaml [hunk 3]
```diff
@@ -54,6 +54,8 @@ spec:
           value: 'true'
         - name: SPRING_CLOUD_DATAFLOW_FEATURES_SCHEDULES_ENABLED
           value: 'true'
+        - name: SPRING_CLOUD_DATAFLOW_TASK_COMPOSED_TASK_RUNNER_URI
+          value: docker://springcloud/spring-cloud-dataflow-composed-task-runner:2.5.0.BUILD-SNAPSHOT
```
Comment (ilayaperumalg) [inline src/kubernetes/server/server-deployment.yaml:58]: @cppwfs I see you also removed the quotes surrounded in the `value`. Is that intentional? Did you try running it on K8s with this change?

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 7c14121cc

**Comment by ilayaperumalg** [comment]
Updated the SCDF CI plan to have the tasks to build and publish the docker images for `spring-cloud-dataflow-composed-task-runner` https://build.spring.io/browse/SCD-BMASTER

**Pull request closed without merging**